--- /tmp/api_client.py	2024-12-18 03:36:59.552742383 +0000
+++ kubernetes_asyncio/client/api_client.py	2024-12-18 03:36:11.062928089 +0000
@@ -535,10 +535,17 @@

         content_types = [x.lower() for x in content_types]

-        if (method == 'PATCH' and
-                'application/json-patch+json' in content_types and
-                isinstance(body, list)):
-            return 'application/json-patch+json'
+        if method == 'PATCH':
+            if ('application/json-patch+json' in content_types and
+                    isinstance(body, list)):
+                return 'application/json-patch+json'
+            if isinstance(body, dict) or hasattr(body, "to_dict"):
+                if 'application/strategic-merge-patch+json' in content_types:
+                    return 'application/strategic-merge-patch+json'
+                elif 'application/merge-patch+json' in content_types:
+                    # Intended for cases where strategic merge patch is not
+                    # supported, like when patching custom objects.
+                    return 'application/merge-patch+json'

         if 'application/json' in content_types or '*/*' in content_types:
             return 'application/json'
