diff --git a/kubernetes_asyncio/client/rest.py b/kubernetes_asyncio/client/rest.py
index e0a33ddb..37b3cf47 100644
--- a/kubernetes_asyncio/client/rest.py
+++ b/kubernetes_asyncio/client/rest.py
@@ -130,7 +130,13 @@ class RESTClientObject(object):
 
         # For `POST`, `PUT`, `PATCH`, `OPTIONS`, `DELETE`
         if method in ['POST', 'PUT', 'PATCH', 'OPTIONS', 'DELETE']:
-            if re.search('json', headers['Content-Type'], re.IGNORECASE):
+            if (
+                re.search("json", headers["Content-Type"], re.IGNORECASE)
+                or headers["Content-Type"] == "application/apply-patch+yaml"
+            ):
+                if headers['Content-Type'] == 'application/json-patch+json':
+                    if not isinstance(body, list):
+                        headers['Content-Type'] = 'application/strategic-merge-patch+json'
                 if body is not None:
                     body = json.dumps(body)
                 args["data"] = body
